src/TreeView/utils.ts

```diff
@@ -317,3 +316,17 @@ export const onComponentBlur = (
     !treeNode.contains(event.nativeEvent.relatedTarget as Node) && callback();
   }
 };
+
+export const isBranchSelectedAndHasSelectedDescendants = (
+  data: INode[],
+  elementId: number,
+  selectedIds: Set<number>
+) => {
+  return (
+    isBranchNode(data, elementId) &&
+    selectedIds.has(elementId) &&
+    [...getDescendants(data, elementId, new Set<number>())].some((item) =>
+      [...selectedIds].includes(item)
```

Comment:
Do we need spread here? getDescendants returns array, selectedIds is Set, you can replace with selectedIds.has(item).

Reply:
you are right, fixed